src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -145,84 +134,85 @@ class SidebarLinks extends React.PureComponent {
      * @param {Object} option
      * @param {String} option.reportID
      */
-    showReportPage(option) {
-        // Prevent opening Report page when clicking LHN row quickly after clicking FAB icon
-        // or when clicking the active LHN row
-        // or when continuously clicking different LHNs, only apply to small screen
-        // since getTopmostReportId always returns on other devices
-        if (this.props.isCreateMenuOpen || this.props.currentReportID === option.reportID || (this.props.isSmallScreenWidth && Navigation.getTopmostReportId())) {
-            return;
-        }
-        Navigation.navigate(ROUTES.getReportRoute(option.reportID));
-        this.props.onLinkClick();
-    }
-
-    render() {
-        return (
-            <View style={[styles.flex1, styles.h100]}>
-                <View
-                    style={[styles.flexRow, styles.ph5, styles.pv3, styles.justifyContentBetween, styles.alignItemsCenter]}
-                    nativeID="drag-area"
-                >
-                    <Header
-                        title={
-                            <LogoComponent
-                                fill={defaultTheme.textLight}
-                                width={variables.lhnLogoWidth}
-                                height={variables.lhnLogoHeight}
-                            />
-                        }
-                        accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
-                        shouldShowEnvironmentBadge
-                    />
-                    <Tooltip text={this.props.translate('common.search')}>
-                        <PressableWithoutFeedback
-                            accessibilityLabel={this.props.translate('sidebarScreen.buttonSearch')}
-                            accessibilityRole={CONST.ACCESSIBILITY_ROLE.BUTTON}
-                            style={[styles.flexRow, styles.ph5]}
-                            onPress={Session.checkIfActionIsAllowed(this.showSearchPage)}
-                        >
-                            <Icon src={Expensicons.MagnifyingGlass} />
-                        </PressableWithoutFeedback>
-                    </Tooltip>
+    const showReportPage = useCallback(
+        (option) => {
+            // Prevent opening Report page when clicking LHN row quickly after clicking FAB icon
+            // or when clicking the active LHN row
+            // or when continuously clicking different LHNs, only apply to small screen
+            // since getTopmostReportId always returns on other devices
+            if (props.isCreateMenuOpen || props.currentReportID === option.reportID || (props.isSmallScreenWidth && Navigation.getTopmostReportId())) {
+                return;
+            }
+            Navigation.navigate(ROUTES.getReportRoute(option.reportID));
+            props.onLinkClick();
+        },
+        [props],
+    );
```

Comment:
creating a `useCallback` that depends on _all_ props defies the purpose of memoizing this function (`useCallback` memoizes this function). (i.e. right now this function would get re-created if any of the props that are not used by this function changes e.g. `localize`).
Make sure to add each prop individually in the [] dependency array.